### PR TITLE
Initial definition of "core" attributes

### PR DESCRIPTION
**Pull request by jskeet**
There may be some more work required to clarify the nature of
extension attributes.

Fixed a few typos I spotted as I went along.

Signed-off-by: Jon Skeet <jonskeet@google.com>

### CODE REVIEW

**Comment by duglin** [comment]
LGTM thanks!

File: cloudevents/primer.md [hunk 1]
```diff
@@ -427,7 +429,7 @@ developers.
 Often extensions are used to test new potential properties of specifications
 prior to them being formally adopted. If there were an `extensions` type of
 property, in which this new property was serialized, then if that property were
-to ever be adopted by the core specification it would be promoted (from a
+to ever be adopted by the core specifcation it would be promoted (from a
```
Comment (duglin) [inline cloudevents/primer.md:432]: missing an "i"

**Comment by duglin** [comment]
Approved on the 4/7 call - will merge once the typo is fixed

File: cloudevents/primer.md (hunk 1 above)
Comment (jskeet) [reply cloudevents/primer.md]: Fixed and repushed, thanks.

**Pull request merged**